### ISSUE TITLE
Installing on iOS app

### ISSUE THREAD

**Issue by KieranLafferty** [Ranks/emojione#363]
Hello,

I have been trying to add the TTF font for iOS/Mac to my app without any success. I am trying to have it override the system emojis within app to use the emojione assets instead of the apple default ones.

I've tried adding AppleColorEmoji@2x.ttf to my project an explicitly setting the font to "AppleColorEmoji" for the particular label. 

Is this possible to do? Thank you in advance for your help!

**Comment by Tmason1** [comment]
I'm having the exact same issue and I don't know how to stop if from getting rid of the original emojis.

**Comment by caseyahenson** [comment]
Hi @KieranLafferty, could you take a look at this tutorial? 

http://codewithchris.com/common-mistakes-with-adding-custom-fonts-to-your-ios-app/

Let me know if this is helpful in getting the font to work in your app.

**Comment by Tmason1** [comment]
It didn't help sorry. I'm trying to change the file name because it
overwrites the original emoji font.

**Comment by KieranLafferty** [comment]
Thanks for the suggestion @caseyahenson. As @Tmason1  mentioned there is a naming conflict with the original AppleColorEmoji font family name which UIFont use the system font instead of the custom ttf. 

Not sure if renaming the font family to something like EmojioneColorEmoji would allow for this to be loaded properly using UIFont.

I've tried renaming the font using FontForge (ttf editing software) but with my limited experience creating/editing fonts wasn't able to get it working properly.

**Comment by Tmason1** [comment]
Even if I change the file name nothing happens. The system font name stays
the same. I've changed it using font forge but none of the characters load
once I've opened it up in the program. When I export it, the name has
changed but all the characters are removed.

**Comment by KieranLaffertySC** [comment]
Yea its not about the file name you actually have to edit the font at the data level. Open up your renamed font in Font Book and it will still say AppleColorEmoji (which is the font family name) this needs to be edited using a program like FontForge to change the font data

**Comment by KieranLafferty** [comment]
Yea its not about the file name you actually have to edit the font at the data level. Open up your renamed font in Font Book and it will still say AppleColorEmoji (which is the font family name) this needs to be edited using a program like FontForge to change the font data

**Comment by Tmason1** [comment]
yeah I've changed it in font forge already but it removes the emoji
characters once I've opened it up in font forge.

**Comment by xoudini** [comment]
@KieranLafferty @Tmason1 
If you're interested, I've recently created a framework in Swift with which you can replace all the standard emoji in strings: https://github.com/xoudini/emojica

It's out-of-the-box compatible with EmojiOne, although you'll just get fallbacks for some of the zero-width joiner sequences since they're not implemented into EmojiOne as of yet.

**Comment by ManuelOrtega** [comment]
Any luck by doing this? I'm struggling trying to integrate Emoji One font in iOS...

**Comment by xoudini** [comment]
@ManuelOrtega – the fonts won't really work as intended in iOS apps, likely because the AppleColorEmoji font overrides any other font when encountering emoji.

However, if you're willing to settle for an image-based solution to replace the standard emoji with EmojiOne, see my above answer. The framework is quite lightweight, and works on `UILabel` and `UITextView`. You can check out the provided example app for usage and how it performs.

**Comment by Tmason1** [comment]
no the problem is with the font, other fonts work. It's because the
EmojiOne's file basename is "AppleColorEmoji". All I need to do is change
it that's all. But i don't know how. Changing name of the file doesn't
change it's base name. That's the only reason why it overrides. Because
they have the same name.

**Comment by ranesr** [comment]
Any luck changing the font name?

**Comment by ranesr** [comment]
@KieranLafferty @Tmason1 I found [this](https://superuser.com/a/1050076) and then I used [TTFEdit](https://sourceforge.net/projects/ttfedit/) to rename the font to this font name - "AppleColorEmojiOne".

And now I can use this new ttf file and include in the project. I did this to use it and it works fine -
```
let font = UIFont(name: "AppleColorEmojiOne", size: 20)
```

I will be happy to make PR to include this file with new font name while keeping the original one in [fonts](https://github.com/Ranks/emojione/tree/master/extras/fonts).

**Comment by Tmason1** [comment]
how do I download it?

**Comment by ranesr** [comment]
Using [this link](https://sourceforge.net/projects/ttfedit/files/latest/download).

**Comment by Tmason1** [comment]
I can't find the font in the files

**Comment by Tmason1** [comment]
it's just a zip file with lots of fonts

**Comment by ranesr** [comment]
You just have to open `ttfedit.jar` after you unzip the downloaded file. Then open [emojione-apple.ttf](https://github.com/Ranks/emojione/blob/master/extras/fonts/emojione-apple.ttf) that you may have already downloaded using ttfedit. It will show you cmap. You will also find name tab vertically on the left side of the window, click on it and you will able to update the family/font names. After that save it as any name you want.


<img width="1434" alt="ttfedit" src="https://cloud.githubusercontent.com/assets/5587292/25871024/b3a15170-34cb-11e7-8a35-d429e990f31e.png">

**Comment by ranesr** [comment]
@KieranLafferty @Tmason1 
I have created [EmojiOne](https://github.com/ranesr/EmojiOne). It currently supports `UIImageView` and `UILabel`.

Until we figure out how to change font name without affecting emoji icons, we can use this library.

**Comment by ltblueberry0** [comment]
Is this ever going to be fixed?

**Comment by czgarrett** [comment]
I think all we need is for EmojiOne to post a new .ttf with a different postscript name.  I was able to get it partially working by opening the font with Glyphs, changing the postscript name and exporting, but then some of the zero width joiner sequences stopped working in my app.  I don't know how to fix them.  I didn't have any luck with FontForge or TTX either.

I think all we need is a .ttf file that is exactly the same as what's on the download page, with the name changed.

**Comment by czgarrett** [comment]
I got it working!  (slightly hacky, but...)

I opened the .ttf file with TextEdit, find/replaced "Emoji" with "Emoj2".  The trick if you edit the binary file directly is that you have to change the font family and postscript name, but keep the names the same length so you don't mess up the file.

If you do that, then you can add the .ttf file to your app in the usual way, and then use "AppleColorEmoj2" as your font name when you look it up with UIFont.
